src/edu/washington/escience/myria/operator/network/distribute/HyperCubeDistributeFunction.java

```diff
@@ -0,0 +1,45 @@
+package edu.washington.escience.myria.operator.network.distribute;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** Distribute function for HyperCube Shuffle. */
+public final class HyperCubeDistributeFunction extends DistributeFunction {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * @param hyperCubeDimensions the sizes of each dimension of the hypercube.
+   * @param hashedColumns which fields are hashed.
+   * @param mappedHCDimensions mapped hypercube dimensions of hashed columns.
+   * @param cellPartition mapping from cells to destinations.
+   */
+  @JsonCreator
+  public HyperCubeDistributeFunction(
+      @JsonProperty("hyperCubeDimensions") final int[] hyperCubeDimensions,
+      @JsonProperty("hashedColumns") final int[] hashedColumns,
+      @JsonProperty("mappedHCDimensions") final int[] mappedHCDimensions,
+      @JsonProperty("callPartition") final List<List<Integer>> cellPartition) {
+    super(
+        new HyperCubePartitionFunction(hyperCubeDimensions, hashedColumns, mappedHCDimensions),
+        cellPartition);
+  }
+
+  /** @return all destinations */
+  public List<Integer> getAllDestinations() {
+    List<Integer> values = new ArrayList<Integer>();
```

Comment:
I agree, however I was following the implementation in https://github.com/uwescience/myria/blob/master/src/edu/washington/escience/myria/api/encoding/HyperShuffleProducerEncoding.java#L57. `cellPartition` contains worker IDs and once flattened and sorted (without dedup), it is expected to be [0,1,...n-1].